spmd/tensor/ops/tp_sharding_ops.py

```diff
@@ -4,40 +4,36 @@
 import torch.utils._pytree as pytree
 from typing import List
 from spmd.tensor.api import DTensor
+from spmd.tensor.dispatch import OpSchema, OutputSharding
 from spmd.tensor.utils import unwrap_local_tensor
-from spmd.tensor.ops.utils import unwrap_single_placement, register_impl
+from spmd.tensor.ops.utils import unwrap_single_placement, register_impl, register_prop_rule
+from spmd.tensor.placement_types import DTensorSpec
 
 """
 The ops below were quickly hacked and needed to be polished down the road.
 Although they come with unit tests already, the logic is directly borrowed
-from ShardedTensor. We need to also make it work for all placement types 
+from ShardedTensor. We need to also make it work for all placement types
 of DTensor and all corner cases for sharded distributed tensor.
 """
 
 
-@register_impl("aten.baddbmm.default")
-def dist_baddbmm(
-    self: DTensor,
-    batch1: DTensor,
-    batch2: DTensor,
-    beta: float = 1.0,
-    alpha: float = 1.0,
-) -> DTensor:
-    local_input, local_batch1, local_batch2 = pytree.tree_map(
-        unwrap_local_tensor, (self, batch1, batch2)
+@register_prop_rule("aten.bmm.default")
+def bmm_rules(op_schema: OpSchema) -> OutputSharding:
+    mat1_spec, mat2_spec = op_schema.args_spec
+    mat1_shape = list(mat1_spec.shape) # b * n * m
+    mat2_shape = list(mat2_spec.shape) # b * m * p
```

Comment:
Sometimes, in backward, we will have a replicate as mat1 and a sharded as mat2. So can you also add logic to handle that. You can reference the code here: https://github.com/pytorch/PiPPy/pull/382/files#diff-44be2c8678cd9e83382ce3e5d09f972b06935f8980bc9410cce9cac0651552d0R143-R150. Thanks!

Reply:
I guess this situation also applies to `baddbmm`, right?

Reply:
Well, theoretically yes, but the backward of baddbmm is bmm. We can add this for baddbmm. Thanks!